micronaut-maven-plugin/src/main/java/io/micronaut/maven/testresources/TestResourcesHelper.java

```diff
@@ -292,9 +290,23 @@ public void stop() throws MojoExecutionException {
         }
     }
 
-    private void doStop() throws IOException {
-        Path settingsDirectory = getServerSettingsDirectory();
-        ServerUtils.stopServer(settingsDirectory);
+    private void doStop() throws IOException, MojoExecutionException {
+        try {
+            Path settingsDirectory = getServerSettingsDirectory();
+            ServerUtils.stopServer(settingsDirectory);
+        } finally {
+            deleteKeepAliveFile();
+        }
+    }
+
+    private void deleteKeepAliveFile() throws MojoExecutionException {
+        if (Files.exists(getKeepAliveFile())) {
+            try {
+                Files.delete(getKeepAliveFile());
+            } catch (IOException e) {
+                throw new MojoExecutionException("Failed to delete keepalive file", e);
```

Comment:
Do we want to fail the build in this case?

Reply:
I would tend to think that if this happens, then it means we have a serious problem, like disk full, gamma rays, ... so failing the build or not doesn't make much of a difference. Note that I only moved the code here, it used to fail already.

Reply:
Fair enough, in case of an alien invasion, the build should not continue 👽